core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -166,6 +166,15 @@ class KryoSerializer(conf: SparkConf)
     kryo.register(Utils.classForName("scala.collection.immutable.$colon$colon"))
     kryo.register(classOf[ArrayBuffer[Any]])
 
+    // Add classes for Streaming
+    try {
+      kryo.register(
+        Utils.classForName("org.apache.spark.streaming.util.OpenHashMapBasedStateMap"),
```

Comment:
Maybe we should add an internal API to let other projects register their classes to KryoSerializer.